manta-crypto/src/algebra.rs

```diff
@@ -113,6 +141,100 @@ impl<G, const N: usize> PrecomputedBaseTable<G, N> {
     }
 }
 
+/// Window Method for Point Multiplication
+pub struct Window<G> {
+    /// Multiplication Table
+    table: Vec<G>,
+}
+
+impl<G> Window<G> {
+    /// Creates a new `Window` from `table` without checking its correctness.
+    #[inline]
+    pub fn new_unchecked(table: Vec<G>) -> Self {
+        Self { table }
+    }
+
+    /// Creates a new `Window`.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if `window_size` is less than `1`.
+    #[inline]
+    pub fn new<COM>(window_size: usize, point: G, compiler: &mut COM) -> Self
+    where
+        G: Group<COM> + Zero<COM>,
+    {
+        assert!(window_size > 0, "Window size must be at least 1.");
+        let table_length = 2usize.pow(window_size as u32);
+        let mut table = Vec::with_capacity(table_length);
+        table.push(G::zero(compiler));
+        for _ in 1..table_length {
+            table.push(table.last().unwrap().add(&point, compiler));
+        }
+        Self::new_unchecked(table)
+    }
+
+    /// Returns the window size.
+    #[inline]
+    pub fn window_size(&self) -> usize {
+        (self.table.len() as f32).log2() as usize
```

Comment:
```suggestion
        self.table.len().trailing_zeroes()
```

Reply:
Done, but isn't this number terribly off if e.g. we have a (possibly unchecked) table length of 2^n + 1?